ompi/mca/common/ompio/common_ompio_file_read.c

```diff
@@ -132,7 +132,7 @@ int mca_common_ompio_file_read (ompio_file_t *fh,
     else {
 	bytes_per_cycle = OMPIO_MCA_GET(fh, cycle_buffer_size);
     }
-    cycles = ceil((float)max_data/bytes_per_cycle);
+    cycles = ceil((double)max_data/bytes_per_cycle);
```

Comment:
please don't change it from
```C
ceil((float)max_data/bytes_per_cycle;
// to
ceil((double)max_data/bytes_per_cycle;
```

which only defers the issue.
Instead, just cast both `max_data` and `bytes_per_cycle` to `uint64_t` to be platform agnostic and convert it to an ["integer-`ceil`"](https://stackoverflow.com/questions/2745074/fast-ceiling-of-an-integer-division-in-c-c) via:
```C
(max_data + bytes_per_cycle - 1) / bytes_per_cycle
```